lib/active_shipping/shipment_packer.rb

```diff
@@ -50,27 +51,32 @@ def pack(items, dimensions, maximum_weight, currency)
 
       private
 
-      def calculate_total_weight(items, maximum_weight)
-        items.inject(0) do |total_weight, item|
+      def validate_total_weight(items, maximum_weight)
+        total_weight = 0
+        items.each do |item|
           total_weight += item[:quantity].to_i * item[:grams].to_i
 
           if overweight_item?(item[:grams], maximum_weight)
             raise OverweightItem, "The item with weight of #{item[:grams]}g is heavier than the allowable package weight of #{maximum_weight}g"
           end
 
-          if excess_package_quantity?(total_weight, maximum_weight)
-            raise ExcessPackageQuantity, "Unable to pack more than #{EXCESS_PACKAGE_QUANTITY_THRESHOLD} packages"
-          end
-
-          total_weight
+          raise_excess_quantity_error if likely_excess_quantity?(total_weight, maximum_weight)
         end
       end
 
+      def validate_package_quantity(number_of_packages)
+        raise_excess_quantity_error if number_of_packages >= EXCESS_PACKAGE_QUANTITY_THRESHOLD
+      end
+
+      def raise_excess_quantity_error
+        raise ExcessPackageQuantity, "Unable to pack more than #{EXCESS_PACKAGE_QUANTITY_THRESHOLD} packages"
+      end
+
       def overweight_item?(grams, maximum_weight)
         grams.to_i > maximum_weight
       end
 
-      def excess_package_quantity?(total_weight, maximum_weight)
+      def likely_excess_quantity?(total_weight, maximum_weight)
```

Comment:
Not sure about the name, since if this returns `true`, it's not _likely_ an excess quantity, it _is_ an excess quantity, correct?
It needs to communicate that even if it returns `false`, it could still be an excess quantity, but I feel like this name communicates the opposite, that even if it returns `true`, it might _not_ be an excess quantity.